config/main.py

```diff
@@ -62,6 +62,44 @@ def log_error(msg):
     syslog.syslog(syslog.LOG_ERR, msg)
     syslog.closelog()
 
+class AbbreviationGroup(click.Group):
+    """This subclass of click.Group supports abbreviated subgroup/subcommand names
+    """
+
+    def get_command(self, ctx, cmd_name):
+        # Try to get builtin commands as normal
+        rv = click.Group.get_command(self, ctx, cmd_name)
+        if rv is not None:
+            return rv
+
+        # Allow automatic abbreviation of the command.  "status" for
+        # instance will match "st".  We only allow that however if
+        # there is only one command.
+        # If there are multiple matches and the shortest one is the common prefix of all the matches, return
+        # the shortest one
+        matches = []
+        shortest = None
+        for x in self.list_commands(ctx):
+            if x.lower().startswith(cmd_name.lower()):
+                matches.append(x)
+                if not shortest:
+                    shortest = x
+                elif len(shortest) > len(x):
+                    shortest = x
+
+        if not matches:
```

Comment:
I notice some code removed during backporting. Any reason?
If valid reason, why not remove on master?

Reply:
What code are you referring to that you claim was removed?

Reply:
Oh I see. You removed the code on master https://github.com/Azure/sonic-utilities/pull/903/files#diff-5f0fe55ee7bf142a05fb98fce9db5b0bL95.

Then you actually backport 2 commits.

Reply:
Kinda. Actually, the other commit was backported before this one (this PR didn't have the "Request for 201911 branch" label); that's why there were conflicts and I needed to open this separately.